### ISSUE TITLE
Allow copying all of Brave Leo conversation

### ISSUE THREAD

**Issue by Segment0895** [brave/brave-browser#42907]
### Description

Since Brave Leo does not save the queries I tend to copy the contents and paste them in my private Joplin instance.

As such, I hereby request that a button is added to allow "copy all" so one can copy both the prompt as the answer in a formatted, organized way.

### Steps to reproduce

1. Open Brave Leo
2. Ask what is the meaning of life
3. Find Brave Leo's answer incredible insightful and try to save it for later.

### Actual result

1. Attempt 1: CTRL+A selects content which does not belong to the conversation
2. Attempt 2: click "copy" on each one of the prompts and fill missing info when pasting

### Expected result

1. Click "copy whole conversation in structured format".
2. Paste with all text OK

### Reproduces how often

Easily reproduced

### Desktop Brave version (brave://version info)

Revision | 45dae32a9fd5d259796f6c975ec055564fac7139
OS | Linux
JavaScript | V8 13.1.201.16
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36
Command Line | /opt/brave.com/brave/brave --disable-domain-reliability --enable-dom-distiller --enable-distillability-service --origin-trial-public-key=SNIP --lso-url=https://no-thanks.invalid --sync-url=https://sync-v2.brave.com/v2 --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater.brave.com/extensions
Executable Path | /opt/brave.com/brave/brave
Profile Path | /home/SNIP/.config/BraveSoftware/Brave-Browser/Profile 1
Active Variations | main@SNIPAllowCertainClientHintsStudy:EnabledBraveAIChatDefaultModelStudy:DefaultLlamaBraveAdblockMobileNotificationsListDefault:EnabledBraveAdsAdEventStudy:EnabledBraveAdsConversionsStudy:EnabledBraveAdsExclusionRulesStudy:EnabledBraveAdsNewTabPageAdsStudy:EnabledBraveAdsRedeemRewardConfirmationStudy:EnabledBraveAdsSiteVisitStudy:EnabledBraveAdsTextClassificationPageProbabilitiesStudy:EnabledBraveAggressiveModeRetirementExperiment:DisabledBraveAutoTranslateStudy:AutoTranslateDisabledBraveDebounceStudy:EnabledBraveFeedUpdateStudy:EnabledBraveForgetFirstPartyStorage:EnabledBraveGoogleSignInPermissionStudy:EnabledBraveHorizontalTabsUpdateEnabledStudy:EnabledBraveP3AConstellationEnabled:EnabledBraveP3AJSONOtherDeprecation:EnabledBraveP3ANebulaRelease:EnabledBraveP3ATypicalJSONDeprecationEnabled:EnabledBraveRewardsAllowSelfCustodyProvidersStudy:EnabledBraveRewardsWebUiPanelStudy:EnabledBraveScreenFingerprintingBlockerStudy:EnabledBraveSearchAdStudy:EnabledBraveWalletAnkrBalancesEnabled:DefaultBraveWebcompatExceptionsServiceReleaseStudy:EnabledClampPlatformVersionClientHint:DisabledCookieListDefaultStudy:EnabledDefault1pBlockingStudy:DisabledDisableReduceLanguage:EnabledEphemeralStorageStudy:EnabledExtensionsManifestV2StudyRelease:EnabledPartitionedCookies:EnabledUndecryptablePasswords:EnabledUserActivityStudy:TriggersWhatsNewStudy:Enabled

### Android device

- Brand/model: Pixel 6a
- Android version: Android 12

### Channel information

- [x] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [x] with Brave Shields disabled
- [x] with Brave Rewards disabled
- [x] in the latest version of Chrome

### Miscellaneous information

this is a feature request, will try to add tag after creating.

**Comment by Segment0895** [comment]
FEATURE REQUEST

**Comment by aguscruiz** [comment]
Designs added here:
https://www.figma.com/design/m0Gdbf0wtqyfEFGm32VLLc/%F0%9F%94%84-Leo--IN-PROGRESS-?node-id=15906-24917&t=Mza15zk2evxpIiJI-1

**Comment by petemill** [comment]
Brave AI _does_ save all your conversations now, so perhaps this is less relevant?

**Comment by aguscruiz** [comment]
Yeah but having a button on the menu like I added would make it much simpler to do.

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.88.110 Chromium: 145.0.7632.76 (Official Build) beta (64-bit)
-- | --
Revision | 4208c41c60ebb44c1c59210e2f22f6c639817bd7
OS | Windows 11 Version 25H2 (Build 26200.7623)
```
Verified the issue using the testplan from https://github.com/brave/brave-core/pull/33375#issue-3844636444

Confirmed 

- Able to open the 3 Dot menu for the conversation and clicked the `Copy entire conversation button`
    - Pasted into a notepad and confirmed that the conversation was copied.
- Able to copy the conversation by clicking `Copy` button at the end of the conversation
  - Pasted into a notepad and confirmed the conversation was copied as expected
  
  https://github.com/user-attachments/assets/fd2f16d6-8609-4c4a-af9a-3b4271a088f2

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 17 CP21.260206.011.A1 release-keys
Brave build: 1.88.124
Chromium: 146.0.7680.32 (Official Build) (64-bit)
```
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-core/pull/33375#issue-3844636444
2. *Verify*

**ACTUAL RESULTS:**
- Verified that Leo's hamburger menu includes the `Copy entire conversation` button.
- Verified that whole conversation can be copied using `Copy entire conversation` button and then pasted in.
- Verified that after tapping `Copy entire conversation` button, confirmation snackbar is shown.

---------------------------

| Dark theme | Light theme |
|:-:|:-:|
| **SCREENCAST** | **SCREENCAST** |
| [![1](https://github.com/user-attachments/assets/88e2ac3a-3829-446f-b864-d87af6353515)](https://github.com/user-attachments/assets/4f65e114-00bf-4fce-ad36-03319ce0729d) | [![2](https://github.com/user-attachments/assets/d0d537bf-fd1a-4068-9410-57cfb08c8c3a)](https://github.com/user-attachments/assets/e042f3cf-132f-4ce7-a75e-d50775d7d724) |
